client/my-sites/checkout/index.js

```diff
@@ -16,13 +16,31 @@ import {
 	conciergeQuickstartSession,
 } from './controller';
 import SiftScience from 'lib/siftscience';
-import { makeLayout, render as clientRender } from 'controller';
+import { makeLayout, redirectLoggedOut, render as clientRender } from 'controller';
 import { noSite, siteSelection } from 'my-sites/controller';
 import config from 'config';
+import userFactory from 'lib/user';
 
 export default function() {
 	SiftScience.recordUser();
 
+	const user = userFactory();
+	const isLoggedOut = ! user.get();
```

Comment:
It's better to use the `isUserLoggedIn` Redux selector here rather than the legacy `lib/user` module (we're trying to get rid of it)

Reply:
This file `client/my-sites/checkout/index.js` does not have a redux connect - do you suggest introducing one for this?

Reply:
Hmm, this is not a React component, so `connect` cannot be used and there is no other way to access the Redux store from the section initialization function 🙁

There's another section that has a mixture of logged-in and logged-out views: [Themes](https://github.com/Automattic/wp-calypso/blob/c50e1d7c7496011e51fa5c114fdc4b9f2c345f3e/client/my-sites/themes/index.web.js#L22-L23). And it uses the `lib/user` module, too. It seems like the only solution right now. We might want to invent some better framework solution later, but that's not for this PR. You can dismiss my earlier comment and leave `lib/user` in there 🙂